src/components/Capture/Document.tsx

```diff
@@ -1,64 +1,97 @@
 import { h, Component } from 'preact'
+
+import { isDesktop, isHybrid, addDeviceRelatedProperties } from '~utils'
+import { validateFile } from '~utils/file'
+import { DOCUMENT_CAPTURE_LOCALES_MAPPING } from '~utils/localesMapping'
+import { randomId } from '~utils/string'
+
 import { appendToTracking } from '../../Tracker'
+import { localised } from '../../locales'
 import DocumentAutoCapture from '../Photo/DocumentAutoCapture'
 import DocumentLiveCapture from '../Photo/DocumentLiveCapture'
 import Uploader from '../Uploader'
 import PageTitle from '../PageTitle'
 import CustomFileInput from '../CustomFileInput'
-import withCrossDeviceWhenNoCamera from './withCrossDeviceWhenNoCamera'
 import { getDocumentTypeGroup } from '../DocumentSelector/documentTypes'
-import { isDesktop, isHybrid, addDeviceRelatedProperties } from '~utils'
-import { compose } from '~utils/func'
-import { validateFile } from '~utils/file'
-import { DOCUMENT_CAPTURE_LOCALES_MAPPING } from '~utils/localesMapping'
-import { randomId } from '~utils/string'
-import { localised } from '../../locales'
 import FallbackButton from '../Button/FallbackButton'
 import theme from '../Theme/style.scss'
+
+import withCrossDeviceWhenNoCamera from './withCrossDeviceWhenNoCamera'
 import style from './style.scss'
 
-class Document extends Component {
-  static defaultProps = {
-    side: 'front',
+import type { ImageResizeInfo } from '~types/commons'
+import type { WithLocalisedProps, WithCaptureVariantProps } from '~types/hocs'
+import type { DocumentCapture } from '~types/redux'
+import type {
+  HandleCaptureProp,
+  StepComponentDocumentProps,
+} from '~types/routers'
+import type { DocumentTypes, PoaTypes } from '~types/steps'
+
+const getDocumentType = (
+  isPoA?: boolean,
+  documentType?: DocumentTypes,
+  poaDocumentType?: PoaTypes
+): DocumentTypes | PoaTypes => {
+  if (isPoA && poaDocumentType) {
+    return poaDocumentType
+  }
+
+  if (documentType) {
+    return documentType
+  }
+
+  throw new Error('Neither documentType nor poaDocumentType provided')
```

Comment:
Both `documentType` and `poaDocumentType` are optional, and we extract the doc type by `isPoA: boolean` which doesn't guarantee either `documentType` and `poaDocumentType` is present. I think this is more about internal fault rather than configurations, so I'll add `trackException` instead.